File: db/csharp/SalesManagement.Domain/Models/Customer.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 顧客マスタ（取引先の顧客としての役割）
    /// </summary>
    public class Customer
    {
        public string CustomerCode { get; set; } = string.Empty;
        public int CustomerBranch { get; set; }
        public int CustomerType { get; set; }
        public string ArCode { get; set; } = string.Empty;
        public int? ArBranch { get; set; }
        public string PayerCode { get; set; } = string.Empty;
        public int? PayerBranch { get; set; }
        public string CustomerName { get; set; } = string.Empty;
        public string? CustomerNameKana { get; set; }
        public string EmployeeCode { get; set; } = string.Empty;
        public string? CustomerUserName { get; set; }
        public string? CustomerDepartmentName { get; set; }
        public string? CustomerZipCode { get; set; }
        public string? CustomerState { get; set; }
        public string? CustomerAddress1 { get; set; }
        public string? CustomerAddress2 { get; set; }
        public string? CustomerTel { get; set; }
        public string? CustomerFax { get; set; }
        public string? CustomerEmail { get; set; }
        public int CustomerArType { get; set; }
        public int CustomerCloseDate1 { get; set; }
        public int CustomerPayMonths1 { get; set; }
        public int? CustomerPayDates1 { get; set; }
        public int CustomerPayMethod1 { get; set; }
        public int CustomerCloseDate2 { get; set; }
        public int CustomerPayMonths2 { get; set; }
        public int? CustomerPayDates2 { get; set; }
        public int CustomerPayMethod2 { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: db/fsharp/SalesManagement.Domain/Models/Customer.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 顧客マスタ（取引先の顧客としての役割）
/// </summary>
[<CLIMutable>]
type Customer = {
    CustomerCode: string
    CustomerBranch: int
    CustomerType: int
    ArCode: string
    ArBranch: int option
    PayerCode: string
    PayerBranch: int option
    CustomerName: string
    CustomerNameKana: string option
    EmployeeCode: string
    CustomerUserName: string option
    CustomerDepartmentName: string option
    CustomerZipCode: string option
    CustomerState: string option
    CustomerAddress1: string option
    CustomerAddress2: string option
    CustomerTel: string option
    CustomerFax: string option
    CustomerEmail: string option
    CustomerArType: int
    CustomerCloseDate1: int
    CustomerPayMonths1: int
    CustomerPayDates1: int option
    CustomerPayMethod1: int
    CustomerCloseDate2: int
    CustomerPayMonths2: int
    CustomerPayDates2: int option
    CustomerPayMethod2: int
    CreatedAt: DateTime
    CreatedBy: string
    UpdatedAt: DateTime
    UpdatedBy: string
}
